In [1]:
import pandas as pd

In [2]:
import sys
stdout = sys.stdout
reload(sys)
sys.setdefaultencoding('utf-8')
sys.stdout = stdout

---

In [3]:
df = pd.read_csv('../reference_w_loc.csv')

In [7]:
df['shape'].unique()

array(['unknown', ' cone', ' light', ' triangle', ' fireball', ' unknown',
       ' rectangle', ' sphere', ' delta', ' disk', ' other', ' changed',
       ' round', ' diamond', ' chevron', ' oval', ' cylinder', ' cigar',
       ' changing', ' egg', ' circle', ' formation', ' pyramid',
       ' crescent', ' flare', ' cross', ' flash', ' teardrop'], dtype=object)

In [8]:
shapes = set(['changed', 'changing', 'chevron', 'cigar', 'circle', 'circular','cone','cresent','cross','cylinder','delta','diamond','dome','disk','egg','fireball','flare','flash','hexagon','light','oval','pyramind','rectangle','rectangular','round','sphere','teardrop','triangle','triangular'])

#### There are 1613 uknown shape entries

In [14]:
df[df['shape'] == "unknown"].shape

(1613, 12)

### Attempt to extract shape from description for unkown shape entries

In [15]:
for index, row in df[df['shape'] == "unknown"].iterrows():
    desc_words = set(row['description'].split())
    shape = row['shape']
    if len(desc_words.intersection(shapes)) > 0:
        df.ix[index, 'shape'] = list(desc_words.intersection(shapes))[0]

In [16]:
df[df['shape'] == "unknown"].shape

(974, 12)

We were able to extract 639 shapes from descriptions, leaving only 974 unkown shape entries

In [18]:
df.to_csv('../reference_w_loc.csv', index=False)